src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -522,6 +537,26 @@ private String getInvoiceLineNumberEndpoint(String id) {
       });
   }
 
+
+  /**
+   * Delete the invoice's poNumbers field, following an invoice line removal.
+   * @param invoice - the invoice of the modified invoice line
+   * @param invoiceLine - the modified invoice line
+   * @param requestContext - used to start new requests
+   */
+  private CompletableFuture<Void> deleteInvoicePoNumbers(Invoice invoice, InvoiceLine invoiceLine, RequestContext requestContext) {
+
+    if (invoiceLine.getPoLineId() == null)
+      return CompletableFuture.completedFuture(null);
+    return orderService.getPoLine(invoiceLine.getPoLineId(), requestContext)
+      .thenCompose(poLine -> orderService.getOrder(poLine.getPurchaseOrderId(), requestContext))
```

Comment:
please use cross-index query for retreiving order by polineId in a single request
`"/orders/composite-orders?query=poLine.id = {{id}}"`